### PR TITLE
Allow users to specify HTTP Header used for API Key auth

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
Shows a field for the HTTP Header name when API Key auth is selected and uses this value as the header name.

### Safety story
This allows users to set the header name which is safe since it is passed to the requests library which in turn uses urllib3 to perform the request.

This code also disables the HTTPS check when running in DEBUG mode which makes it possible to test integrations with other locally run services.

### Automated test coverage
None

### QA Plan
I did some basic QA on my local machine creating connection settings and using them.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Robert-Costello — APPROVED** [review]

**Review by pxwxnvermx — APPROVED** [review]

**Comment by snopoke** [comment]
@pxwxnvermx @Robert-Costello updated with lint changes in the last commit

File: corehq/motech/static/motech/js/connection_settings_detail.js [hunk 1]
```diff
@@ -62,6 +63,7 @@ hqDefine("motech/js/connection_settings_detail", [
                     break;
                 case 'api_key':
                     visible = {
+                        'username': gettext("HTTP Header Name"),
```
Comment (millerdev) [inline corehq/motech/static/motech/js/connection_settings_detail.js:66]: What is the UX in the default use case? Is this field expected to be left blank or is the default value of "Authorization" auto-populated? If blank, is the expectation obvious, or will the user likely feel compelled to enter a value?

Comment (snopoke) [reply corehq/motech/static/motech/js/connection_settings_detail.js:66]: Good question. The default is blank and if left blank the "Authorization" will be used. I did want to make that the default in the UI but it's a bit janky due to how the fields are reused between different auth types.

Given how infrequently this is used and that if you are using API Key auth you'd almost certainly know which header to use (or know where to look it up in your service docs) I decided to leave it.

But open to push back.

Comment (millerdev) [reply corehq/motech/static/motech/js/connection_settings_detail.js:66]: Not a great fix, but maybe a note in the field name would help? I don't love my choice of words, so feel free to change if you can think of something better.
```suggestion
                        'username': gettext("HTTP Header Name (leave empty for default 'Authorization')"),
```

Comment (snopoke) [reply corehq/motech/static/motech/js/connection_settings_detail.js:68]: I've updated to as follows:

* clear the field value when the field label changes e.g. from `Username` to `HTTP Header`
* Set the `placeholder` for `HTTP Header` to `Authorization`

**Review by millerdev — APPROVED** [review]

**Pull request merged**